$MirOS: ports/net/tor/patches/patch-src_or_config_c,v 1.2 2007/09/07 21:40:37 tg Exp $
--- src/or/config.c.orig	Mon Jan  7 07:54:05 2008
+++ src/or/config.c	Tue Mar  4 01:20:32 2008
@@ -2642,15 +2642,15 @@ options_validate(or_options_t *old_optio
 
   if (options->BandwidthRate > ROUTER_MAX_DECLARED_BANDWIDTH) {
     r = tor_snprintf(buf, sizeof(buf),
-                     "BandwidthRate must be at most %d",
-                     ROUTER_MAX_DECLARED_BANDWIDTH);
+                     "BandwidthRate must be at most %ld",
+                     (long)ROUTER_MAX_DECLARED_BANDWIDTH);
     *msg = tor_strdup(r >= 0 ? buf : "internal error");
     return -1;
   }
   if (options->BandwidthBurst > ROUTER_MAX_DECLARED_BANDWIDTH) {
     r = tor_snprintf(buf, sizeof(buf),
-                     "BandwidthBurst must be at most %d",
-                     ROUTER_MAX_DECLARED_BANDWIDTH);
+                     "BandwidthBurst must be at most %ld",
+                     (long)ROUTER_MAX_DECLARED_BANDWIDTH);
     *msg = tor_strdup(r >= 0 ? buf : "internal error");
     return -1;
   }
